adx/resource_adx_external_table.go

```diff
@@ -0,0 +1,270 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type ADXExternalTable struct {
+	Name              string
+	ConnectionStrings string
+	Partitions        string
+	PathFormat        string
+	Folder            string
+	Properties        string
+}
+
+func resourceADXExternalTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceADXExternalTableCreateUpdate,
+		UpdateContext: resourceADXExternalTableCreateUpdate,
+		ReadContext:   resourceADXExternalTableRead,
+		DeleteContext: resourceADXExternalTableDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster": getClusterConfigInputSchema(),
+			"database_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"data_format": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"storage_connection_string": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"schema": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"kind": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "storage",
+			},
+
+			"partitions": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "jsonencode([])",
+			},
+
+			"path_format": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"folder": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"doc_string": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"compressed": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"include_headers": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "All",
+			},
+
+			"name_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"file_extension": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"encoding": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "UTF8NoBOM",
+			},
+
+			"sample_uris": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"files_preview": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"validate_not_empty": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"dry_run": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+		CustomizeDiff: clusterConfigCustomDiff,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
```

Comment:
Does this need to have a 30 *minutes* timeout? In certain cases TF will literally wait for 30 mins before failing.

Reply:
No, not really. Copy pasted it from another resource. What time do you suggest?